internal/thelma/cli/commands/bees/apply_schedule/apply_schedule_command.go

```diff
@@ -0,0 +1,231 @@
+package apply_schedule
+
+import (
+	"fmt"
+	"github.com/broadinstitute/thelma/internal/thelma/app"
+	"github.com/broadinstitute/thelma/internal/thelma/cli"
+	"github.com/broadinstitute/thelma/internal/thelma/cli/commands/bee/common/builders"
+	"github.com/broadinstitute/thelma/internal/thelma/cli/commands/bee/common/filterflags"
+	"github.com/broadinstitute/thelma/internal/thelma/cli/commands/bee/common/views"
+	"github.com/broadinstitute/thelma/internal/thelma/state/api/terra"
+	"github.com/broadinstitute/thelma/internal/thelma/toolbox/argocd"
+	"github.com/broadinstitute/thelma/internal/thelma/utils/pool"
+	"github.com/broadinstitute/thelma/internal/thelma/utils/schedule"
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+	"sync"
+	"time"
+)
+
+const helpMessage = `Start and stop BEEs as defined by their schedule`
+
+type options struct {
+	dryRun      bool
+	start       bool
+	stop        bool
+	fromPast    time.Duration
+	maxParallel int
+}
+
+var flagNames = struct {
+	dryRun      string
+	start       string
+	stop        string
+	fromPast    string
+	maxParallel string
+}{
+	dryRun:      "dry-run",
+	start:       "start",
+	stop:        "stop",
+	fromPast:    "from-past",
+	maxParallel: "max-parallel",
+}
+
+type command struct {
+	options options
+	fflags  filterflags.FilterFlags
+}
+
+func NewBeesApplyScheduleCommand() cli.ThelmaCommand {
+	return &command{
+		fflags: filterflags.NewFilterFlags(),
+	}
+}
+
+func (cmd *command) ConfigureCobra(cobraCommand *cobra.Command) {
+	cobraCommand.Use = "apply-schedule"
+	cobraCommand.Short = helpMessage
+	cobraCommand.Long = helpMessage
+
+	cobraCommand.Flags().BoolVar(&cmd.options.dryRun, flagNames.dryRun, true, "Print the actions that would be taken instead of also doing them")
+	cobraCommand.Flags().BoolVar(&cmd.options.start, flagNames.start, true, "If start schedules should be applied")
+	cobraCommand.Flags().BoolVar(&cmd.options.stop, flagNames.stop, true, "if stop schedules should be applied")
+	cobraCommand.Flags().DurationVar(&cmd.options.fromPast, flagNames.fromPast, 20*time.Minute, "How far back to look for schedule transitions to apply")
+	cobraCommand.Flags().IntVar(&cmd.options.maxParallel, flagNames.maxParallel, 3, "Number of BEEs to apply schedules for in parallel")
+
+	cmd.fflags.AddFlags(cobraCommand)
+}
+
+func (cmd *command) PreRun(_ app.ThelmaApp, _ cli.RunContext) error {
+	return nil
+}
+
+func (cmd *command) Run(app app.ThelmaApp, rc cli.RunContext) error {
+	bees, err := builders.NewBees(app)
+	if err != nil {
+		return err
+	}
+
+	beeFilter, err := cmd.fflags.GetFilter(app)
+	if err != nil {
+		return err
+	}
+
+	matchingBees, err := bees.FilterBees(beeFilter)
+	if err != nil {
+		return err
+	}
+
+	if len(matchingBees) == 0 {
+		log.Info().Msg("Found no matching BEEs to check scheduling for")
+		return nil
+	}
+
+	now := time.Now()
+	since := now.Add(-cmd.options.fromPast)
```

Comment:
Why not `time.Sub`?

Reply:
Because I thought it didn't exist at first, whoops, good catch

Reply:
Oh no actually I'm correct--time.sub accepts another time as argument and returns a duration. We want duration as argument returning time, which is the annoying add-minus thing I found online.